src/SwipeableDrawer/SwipeArea.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import withStyles from '../styles/withStyles';
+import { capitalize } from '../utils/helpers';
+import { isHorizontal } from '../Drawer/Drawer';
+
+export const styles = theme => ({
+  discovery: {
+    position: 'fixed',
+    top: 0,
+    height: '100vh',
+    zIndex: theme.zIndex.drawer - 1,
+  },
+  discoveryAnchorLeft: {
+    left: 0,
+    right: 'auto',
+  },
+  discoveryAnchorRight: {
+    left: 'auto',
+    right: 0,
+  },
+  discoveryAnchorTop: {
+    top: 0,
+    left: 0,
+    bottom: 'auto',
+    right: 0,
+  },
+  discoveryAnchorBottom: {
+    top: 'auto',
+    left: 0,
+    bottom: 0,
+    right: 0,
+  },
+});
+
+function SwipeArea(props) {
```

Comment:
I wasn't able to inline this into `SwipeableDrawer` because adding `withStyles` to it would break just pushing the styles through to the underlying `Drawer`.